### PR TITLE
Custom installation path support

### PR DESCRIPTION
**Pull request by Pythyu**
### Description

Allow omnibus to build an agent able to be ran from any path by adding the origin flag inside of linux `LDFLAGS` and adds two libraries inside of the healthcheck whitelist for macOS.

### CODE REVIEW

File: lib/omnibus/health_check.rb [hunk 1]
```diff
@@ -135,6 +135,8 @@ class HealthCheck
       /libSystem\.B\.dylib/,
       # Symlink of the previous one
       /libgcc_s\.1\.dylib/,
+      /libz\.1\.dylib/,
+      /libdatadog-agent-rtloader\.1\.dylib/,
```
Comment (alopezz) [inline lib/omnibus/health_check.rb]: Context is missing for this.

1. It might be fine for `libz` to be whitelisted; this just means we're fine with _potentially_ relying on the version shipped with macOS, though I think it should be a conscious decision and avoid shipping it if we're planning on relying on the system version. At any rate, it's not clear how this is related to the task at hand.
2. For libdatadog-agent-rtloader, I doubt this is a fine thing to whitelist; but at any rate I really think this should not be whitelisted inside omnibus-ruby, if anything it should live close to where this is actually built.

File: lib/omnibus/software.rb [hunk 2]
```diff
@@ -741,7 +741,7 @@ def with_standard_compiler_flags(env = {}, opts = {})
           elsif platform_version.satisfies?(">= 5.11")
             solaris_flags = {
               "CC" => "gcc -m64 -static-libgcc",
-              "LDFLAGS" => "-Wl,-rpath,#{install_dir}/embedded/lib -L#{install_dir}/embedded/lib -static-libgcc",
```
Comment (alopezz) [inline lib/omnibus/software.rb]: Why modify solaris when we don't for example touch a few other unices that we don't really support?

File: lib/omnibus/software.rb [hunk 3]
```diff
@@ -786,7 +786,7 @@ def with_standard_compiler_flags(env = {}, opts = {})
           }
         else
           {
-            "LDFLAGS" => "-Wl,-rpath,#{install_dir}/embedded/lib -L#{install_dir}/embedded/lib",
+            "LDFLAGS" => "-Wl,-rpath,#{install_dir}/embedded/lib,-z,origin -L#{install_dir}/embedded/lib",
```
Comment (alopezz) [inline lib/omnibus/software.rb:790]: Might be adding a comment about what `-z origin` does (at the very least worth a mention on the PR description).

**Review by alopezz — CHANGES_REQUESTED** [review]

File: lib/omnibus/health_check.rb (hunk 1 above)
Comment (Pythyu) [reply lib/omnibus/health_check.rb]: After investigation we realized that the health check isn't interpreting `@loader_path`when inside of `@rpath` causing the issue. Fix of the health check in progress + removal both of these lib inside of the whitelist

File: lib/omnibus/software.rb (hunk 2 above)
Comment (Pythyu) [reply lib/omnibus/software.rb]: 👍

File: lib/omnibus/health_check.rb [hunk 4]
```diff
@@ -552,11 +553,11 @@ def check_for_bad_macos_library(bad_libs, current_library, name, linked)
           #         name /opt/datadog-agent/embedded/lib (offset 12)
           yield_shellout_results("otool -l #{current_library} | grep LC_RPATH -A2 | grep path | awk '{ print $2 }'") do |rpath|
             # The rpath variable contains a \n (\r\n on Windows), so we remove it when including it in the complete path
-            possible_paths << linked.sub("@rpath", rpath.chop)
+            # If @rpath contains @loader_path we're replacing it with it's actual loader_path
```
Comment (alopezz) [inline lib/omnibus/health_check.rb]: Just a suggestion for more concise wording, feel free to ignore.
```suggestion
            # We also resolve possible `@loader_path` references
```

**Review by alopezz — APPROVED** [review]

**Review by chouquette — APPROVED** [review]
